conduct.md

```diff
@@ -10,11 +10,8 @@ redirect_from:
 
 ## Scala Code of Conduct
 
-We are committed to providing a friendly, safe and welcoming environment for
-all, regardless of age, body size, disability, ethnicity, sex characteristics,
-gender identity and expression, level of experience, education, socio-economic
-status, nationality, personal appearance, race, religion, sexual identity
-and orientation, or other such characteristics.
+We are committed to providing a friendly, safe and welcoming environment.
+We expect the following standards to be observed and upheld by all participants in any community forum (Discuss, GitHub, gitter, etc.).
 
```

Comment:
There are multiple distinct changes happening in this part of the diff.

I like having "We are committed to providing a friendly, safe and welcoming environment" stand as a sentence by itself.

I am _not_ comfortable with reducing the emphasis on diversity so sharply. 

I don't understand why it's important to include a list of forums here (the "Discuss, GitHub, gitter" parenthetical). Why is it important to foreground that so prominently? (If we do include it, s/Discuss/Discourse` and `s/gitter/Gitter`. But I suggest we don't include it here. If it's missing from the document as a whole, is there a place further down it could go?)

Reply:
It wasn't removed, just moved to the "kind and courteous" section.  Do you mean that you think it should be at the top for emphasis?  (I don't think it works all that well in the "encouraged" section TBH.)

Reply:
> It wasn't removed, just moved to the "kind and courteous" section. Do you mean that you think it should be at the top for emphasis?

Yes, that's what I meant.

Reply:
> I don't think it works all that well in the "encouraged" section TBH

and yes, that's a good additional point. I was primarily objecting to the reduction in emphasis, but even apart from that, I agree that the new location doesn't work

Reply:
After reading the discussion point from @rkuhn I understand the goal of the move: get the friendly stuff up, as that presents a more welcoming environment than a list of rules that we're scrutinizing you for.  (If you want to be friendly, "show, don't tell".  We should sound friendly!)

However, I don't think it works very well as implemented.  I think we can keep the positive stuff up front; on reflection it does sound awfully hostile and/or scary in the first few lines.  ("You're gonna discriminate against someone on the basis of personal appearance, aren't you?  Aren't you!  Don't you try it, buddy, we're watching your every move, with our fingers over the ban button." or "We say we're welcoming, but TBH all this horrible stuff happens every day on our forums, and we're soooooo fed up with it!  But, uh, you, um, will be fine, totally fine here now, cause uh, yeah, it's all good from now on; how are you?  (Fires blaster into mic.)")

We can reflect that we're serious about being welcoming to people of diverse backgrounds (who are often left out of "all" when being welcomed) by strengthening our wording later on when we've already established, by acting the part, that we're serious about being welcoming.  Then readers are more likely to feel that we've welcomed them in, and these things are as protections to keep things welcoming in practice.

We just need to find the right spot for it.

Reply:
Again, @Ichoran is spot on, and I agree that the current placement is not ideal — I copied it from the Idris community standards. I’ll try out different placements and make a new suggestion.

Reply:
@SethTisue The list of forums was also inspired by the Idris community standard, but I agree that it detracts from the main focus. How about the updated proposal that only reminds people that there are online and offline forums?

Reply:
@SethTisue Can we consider this resolved?

Reply:
~~regardless of age, body size, disability, ethnicity, sex characteristics,	We expect the following standards to be observed and upheld by all participants in any community forum, be that online or offline.
gender identity and expression, level of experience, education, socio-economic	
status, nationality, personal appearance, race, religion, sexual identity	
and orientation, or other such characteristics.~~

Reply:
@rkuhn no, I'm afraid not. I repeat:

I am _not_ comfortable with reducing the emphasis on diversity so sharply.

In the original text, it's prominent up front that we care about this. in the new text, it's.... not _completely_ buried, but is certainly substantially less prominent than it was. I oppose this strongly.

I do not think the current placement is hostile or scary.

I definitely do not agree at all that this change can be characterized as making the CoC "more friendly". I think it makes it less friendly for groups of people who are underrepresented and discriminated against in tech. Being clear that we want Scala to be a friendly place for those people is one of the main goals of this code.

Reply:
Can we switch the text under "our standards" to be first?  I think it's important to have a positive, uplifting message as the first full thought, and a list of "regardless of" groups isn't very effective at doing that.  If you want people to not worry about age discrimination, mentioning that you will accept them regardless of age _still makes them think about age in the context of discrimination_ which is not such a positive thing.

I don't think it needs to be pushed all the way into the don't-do-this category, as it can be phrased kinda-positively (and was in the original).

But as the very first thing we encounter, as a positive opening statement, it isn't very good.

Reply:
> I don't think it needs to be pushed all the way into the don't-do-this category, as it can be phrased kinda-positively (and was in the original).

Agreed. Let's not make it a list of negatives. We could add another sentence before it that's more active, positive, and welcoming. Such a sentence could build motivation for the list. At present, it come up abruptly, but it would feel more natural if we made it more obvious that this opening paragraph is not about rules, but our goals/mission (things like expanding the community and providing space where members can flourish, etc).

> list of forums

I agree with Seth that it's best to leave this open. It's not just online and offline forums either. Conferences, meetups, and other events use this Code of Conduct. Typelevel and ny-scala use it verbatim.

@rkuhn  Consider for a second the audience for this piece. Who really needs to see this? Who is actively looking for the Code of Conduct? If that person is looking for this list, it should probably be closer to the beginning. It should read more like a goal and less like a rule or afterthought.

Reply:
If we want to keep an emphasis on us valuing diversity, maybe instead of the long list, we can say so explicitly here, in the form of "We are committed to providing a friendly, safe and welcoming environment that is accepting of diversity."

If we value diversity, we should at least provide an accepting environment, and say so explicitly.

Reply:
Numerous valid concerns have been raised in regard to putting the list at the top in its current form. The concern for emphasis of diversity, may it be given a high priority, has to be incorporated into the argumentation that went before instead of simply dismissing the same.

If there are reasons to dismiss previous argumentation, or add to the same, those reasons themselves need to be communicated. As soon as the authority for what is permissible is based what one person or another feels comfortable with, instead of sound reason that can be displayed, justified and understood, a collaborative due process is no longer possible.

Reply:
@otto-dev - If the list were the only thing that was improved here, then we should have a deeper discussion here about the content.  But it isn't the only thing, and that discussion is potentially a long one, so it's probably more effective to conservatively consolidate the other improvements we've made so far and save the real discussion of the list for yet another PR or issue.

You've made a reasonable point about lists potentially putting off people by what they include or exclude, and I see no evidence that this has been given careful consideration.  It should be.

However, ultimately I think there is not anywhere near enough evidence to unambiguously support a particular course of action, especially in light of people typically having a non-identical set of core values.  (Despite being non-identical, people's values are typically quite similar; the similarities tend to go unstated, leading us to think that they're rather different.  Operationally they're usually not.  But it can be enough to matter.)  So the point of creating robust argumentation cannot here be to deliver a decision that is independent of personal opinion.  Nonetheless, the opinion(s) can be informed by careful reasoning and incorporation of a variety of perspectives, rather than whatever people happened to feel about it before careful consideration.  (What they feel also is probably due in part to reasoning and consideration of some perspectives, but perhaps not as many as would be appropriate.)  And the reasoning thought to be most important in this case can be spelled out rather than leaving it ambiguous, implicit, or otherwise unstated.

Reply:
Thanks, and agreed. I don't expect an unbiased outcome, but that there is an active *effort* to facilitate an unbiased process. The way this issue is handled thus far makes me question if possible biases that are contained in the document itself will be removed. Anyway, the future will show, let's leave this discussion alone, and focus on the issues at hand.

Reply:
I’ve reverted the change to the preamble, to be improved in a follow-up PR.